src/dos/dos_keyboard_layout.cpp

```diff
@@ -1243,6 +1244,164 @@ const char* DOS_GetLoadedLayout(void) {
+	{"ro",     Country::Romania        }, // Standard
+	{"ro446",  Country::Romania        }, // QWERTY
+	{"rh",     Country::United_States  }, // Right-Hand Dvorak
+	{"ru",     Country::Russia         }, // Standard
+	{"ru443",  Country::Russia         }, // Typewriter
+	{"rx",     Country::Russia         }, // Extended Standard
+	{"rx443",  Country::Russia         }, // Extended Typewriter
+	{"sd",     Country::Switzerland    }, // German
+	{"sf",     Country::Switzerland    }, // French
+	{"sg",     Country::Switzerland    }, // German
+	{"si",     Country::Slovenia       },
+	{"sk",     Country::Czech_Slovak   }, // Slovakia
+	{"sp",     Country::Spain          },
+	{"sq",     Country::Albania        }, // No-deadkeys
+	{"sq448",  Country::Albania        }, // Deadkeys
+	{"sr",     Country::Serbia         }, // Deadkey
+	{"su",     Country::Finland        },
+	{"sv",     Country::Sweden         },
+	{"sx",     Country::Spain          }, // International
+	{"tj",     Country::Tadjikistan    },
+	{"tm",     Country::Turkmenistan   },
+	{"tr",     Country::Turkey         }, // QWERTY
+	{"tr440",  Country::Turkey         }, // Non-standard
+	{"tt",     Country::Russia         }, // Tatarstan Standard
+	{"tt443",  Country::Russia         }, // Tatarstan Typewriter
+	{"ua",     Country::Ukraine        }, // 101-key
+	{"uk",     Country::United_Kingdom }, // Standard
+	{"uk168",  Country::United_Kingdom }, // Allternate
+	{"ur",     Country::Ukraine        }, // 101-key
+	{"ur465",  Country::Ukraine        }, // 101-key
+	{"ur1996", Country::Ukraine        }, // 101-key
+	{"ur2001", Country::Ukraine        }, // 102-key
+	{"ur2007", Country::Ukraine        }, // 102-key
+	{"us",     Country::United_States  }, // Standard
+	{"ux",     Country::United_States  }, // International
+	{"uz",     Country::Uzbekistan     },
+	{"vi",     Country::Vietnam        },
+	{"yc",     Country::Serbia         }, // Deadkey
+	{"yc450",  Country::Serbia         }, // No-deadkey
+	{"yu",     Country::Yugoslavia     },
+        // clang-format on
+};
+
+static bool country_number_exists(const int requested_number)
+{
+	for ([[maybe_unused]] const auto &[code, number] : country_code_map)
+		if (requested_number == static_cast<int>(number))
+			return true;
+	return false;
+}
+
+static Country lookup_country_from_code(const char *country_code)
+{
+	if (country_code)
+		if (const auto it = country_code_map.find(country_code);
+		    it != country_code_map.end())
+			return it->second;
+	return Country::United_States;
+}
+
+[[maybe_unused]] static uint16_t lookup_codepage_from_country(const Country country)
+{
+	switch (country) {
+	// clang-format off
+	case Country::Finland: return 850;
+	case Country::France:  return 858;
+	case Country::Germany: return 858;
+	case Country::Italy:   return 858;
+	case Country::Poland:  return 852;
+	case Country::Russia:  return 866;
+	case Country::Spain:   return 858;
+	default:               return 437;
+	// clang-format off
```

Comment:
Hungarian definitely uses 852, so that could be added.

[Code page 852](https://en.wikipedia.org/wiki/Code_page_852) (DOS-Latin 2) is used for a bunch of other European Latin-alphabet languages with extra diacritical marks, so you can map all of these to 852 too:

- Bosnian, Croatian, Czech, Hungarian, Polish (already there), Romanian, Serbian, Slovak or Slovene

Some further low-hanging fruits worth adding:

- 857 - Turkish
- 861 - Icelandic
- 775 - Estonian, Lithuanian, Latvian
- 865 - Danish, Norwegian

I'm fairly certain about these; for other Latin-alphabet languages there appear to be multiple commonly used options (e.g. two different code pages for Portugese in Portugal vs Brazil). Non-Latin alphabets are another story; probably those code pages were only used when editing text in text-editors as they lack Latin letters, thus you wouldn't even be able to type DOS commands or read a directory listing!

Reply:
Added (fully list in comments below)